pkg/controller/replicaset/replica_set.go

```diff
@@ -671,9 +694,8 @@ func (rsc *ReplicaSetController) syncReplicaSet(key string) error {
 	// Always updates status as pods come up or die.
 	if err := updateReplicaCount(rsc.kubeClient.Extensions().ReplicaSets(rs.Namespace), rs, len(filteredPods), fullyLabeledReplicasCount, readyReplicasCount); err != nil {
 		// Multiple things could lead to this update failing. Requeuing the replica set ensures
-		// we retry with some fairness.
-		glog.V(2).Infof("Failed to update replica count for controller %v/%v; requeuing; error: %v", rs.Namespace, rs.Name, err)
-		rsc.enqueueReplicaSet(&rs)
+		// Returning an error causes a requeue without forcing a hotloop
+		errlist = append(errlist, err)
 	}
-	return nil
+	return utilerrors.NewAggregate(errlist)
```

Comment:
The RC controller is not aggregating errors but instead it will return them separately: there are only two of them anyway, one error representing a replica create/delete failure and one about a failed RC status update. In the first case it will retry forever albeit being rate-limited and in the second case it will only retry once more. There is no reason for not having both controllers being consistent as far as error handling is concerned.  I am not sure though if the RC controller is 100% correct.